
From: Eric W. Biederman <ebiederm@xmission.com

The following patch simply adds a shutdown method to the x86_64 i8259 code.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/x86_64/kernel/i8259.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+)

diff -puN arch/x86_64/kernel/i8259.c~kexec-i8259-shutdown-x86_64 arch/x86_64/kernel/i8259.c
--- 25/arch/x86_64/kernel/i8259.c~kexec-i8259-shutdown-x86_64	2004-08-22 21:40:43.048858832 -0700
+++ 25-akpm/arch/x86_64/kernel/i8259.c	2004-08-22 21:40:43.051858376 -0700
@@ -416,10 +416,24 @@ static int i8259A_suspend(struct sys_dev
 	return 0;
 }
 
+
+
+static int i8259A_shutdown(struct sys_device *dev)
+{
+	/* Put the i8259A into a quiescent state that
+	 * the kernel initialization code can get it
+	 * out of.
+	 */
+	outb(0xff, 0x21);	/* mask all of 8259A-1 */
+	outb(0xff, 0xA1);	/* mask all of 8259A-1 */
+	return 0;
+}
+
 static struct sysdev_class i8259_sysdev_class = {
 	set_kset_name("i8259"),
 	.suspend = i8259A_suspend,
 	.resume = i8259A_resume,
+	.shutdown = i8259A_shutdown,
 };
 
 static struct sys_device device_i8259A = {
_
